File: assemblies/NMaven.Artifact/src/main/csharp/NMaven/Artifact/PathUtil.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text;

namespace NMaven.Artifact
{
    public class PathUtil
    {
        public static FileInfo GetPrivateApplicationBaseFileFor(Artifact artifact, DirectoryInfo localRepository)
        {
            return new FileInfo(localRepository.Parent.FullName + @"\pab\gac_msil\" + artifact.ArtifactId + @"\" + artifact.Version + "__" +
                artifact.GroupId + @"\" + artifact.ArtifactId + "." + artifact.Extension);
        }

        public static FileInfo GetUserAssemblyCacheFileFor(Artifact artifact, DirectoryInfo localRepository)
        {
            return new FileInfo(localRepository.Parent.FullName + @"\uac\gac_msil\" + artifact.ArtifactId + @"\" + artifact.Version + "__" +
                artifact.GroupId + @"\" + artifact.ArtifactId + "." + artifact.Extension);
        }
    }
}

File: components/dotnet-artifact/src/main/java/org/apache/maven/dotnet/artifact/PathUtil.java
package org.apache.maven.dotnet.artifact;

import org.apache.maven.artifact.Artifact;
import org.apache.maven.dotnet.ArtifactType;

import java.io.File;

public class PathUtil
{
    public static File getPrivateApplicationBaseFileFor( Artifact artifact, File localRepository )
    {
        return new File( localRepository.getParentFile(), "pab\\gac_msil\\" + artifact.getArtifactId() + "\\" +
            artifact.getVersion() + "__" + artifact.getGroupId() + "\\" + artifact.getArtifactId() + "." +
            ArtifactType.getArtifactTypeForPackagingName( artifact.getType() ).getExtension() );
    }

    public static File getUserAssemblyCacheFileFor( Artifact artifact, File localRepository )
    {
        return new File( localRepository.getParentFile(), "uac\\gac_msil\\" + artifact.getArtifactId() + "\\" +
            artifact.getVersion() + "__" + artifact.getGroupId() + "\\" + artifact.getArtifactId() + "." +
            ArtifactType.getArtifactTypeForPackagingName( artifact.getType() ).getExtension() );
    }
}
